### DISCUSSION TITLE
Bridge GeoPoint to CLLocationCoordinate2D

### DISCUSSION THREAD

**Discussion by michaelscampbell** [firebase/firebase-ios-sdk#9958] [Ideas]
I notice that the Firebase libraries bridge some types from Firebase to Swift. For example, Timestamp to Date. It would be helpful to me for Firebase to bridge GeoPoint to CLLocationCoordinate2D also.

In particular, when converting a Firebase document to a custom object using `documentReference.document.data(as: type.self)` I would like to be able to convert a Firebase GeoPoint field directly to a CLLocationCoordinate2D as part of the conversion process. 

Today I have to store a GeoPoint in my model. I then have a computed variable of type  CLLocationCoordinate2D based on that GeoPoint. The rest of my app references the CLLocationCoordinate2D. So this conversion would allow me to simplify my model in multiple places (each object that references a GeoPoint).

Both types store a latitude and longitude so the conversion between them might be straight forward.

**Comment by paulb777** [comment]
Thanks for starting the discussion. I'm going to transfer this to a feature request for better visibility and tracking
